### ISSUE TITLE
How to use createNewItemRenderer with custom itemListRenderer

### ISSUE THREAD

**Issue by dimkajasons** [palantir/blueprint#4031]
#### Environment

- __Package version(s)__:  "@blueprintjs/select": "3.11.1",
- __Browser and OS versions__: Chrome Version 80.0.3987.132, macOS Catalina 10.15.3

#### Question

I am trying to use MultiSelect component with custom itemListRenderer and createNewItemFromQuery.
But in my callback function for render Query List I haven't renderer for create new item

**Comment by justinbhopper** [comment]
@dimkajasons You are supposed to use `createNewItemRenderer` to render the `createNewItemFromQuery` item.  The documentation illustrates this quite well:

https://blueprintjs.com/docs/#select/select-component.creating-new-items

**Comment by gigerdo** [comment]
I have the same problem and the documentation does not illustrate this case. It only illustrates how to have a create-new-entry button with the default item-list-renderer.

Looking at the code it is necessary to recreate quite a bit of code that is included in the default item-list-renderer:
```
/** default itemListRenderer implementation */
    private renderItemList = (listProps: IItemListRendererProps<T>) => {
        const { initialContent, noResults } = this.props;

        // omit noResults if createNewItemFromQuery and createNewItemRenderer are both supplied, and query is not empty
        const maybeNoResults = this.isCreateItemRendered() ? null : noResults;
        const menuContent = renderFilteredItems(listProps, maybeNoResults, initialContent);
        const createItemView = this.isCreateItemRendered()
            ? this.renderCreateItemMenuItem(this.state.query.trim())
            : null;
        if (menuContent == null && createItemView == null) {
            return null;
        }
        return (
            <Menu ulRef={listProps.itemsParentRef}>
                {menuContent}
                {createItemView}
            </Menu>
        );
    };
```

However this code accesses internal code to select the newly created item and then close the popover. I'm not sure how to replicate this in my own item-list-renderer. 

Basically I have to call my own createNewItemRenderer with all the necessary properties. However the 'handleClick' function is provided internally by the Select and I think there is no way to access it from outside.

Does someone know how this could be done?

**Comment by gigerdo** [comment]
I built a custom itemListRenderer supporting this feature, but it seems hacky. It works but it would be cool if there is a more elegant way to do it.

```
const selectRef = useRef<any>(null); // Pass this ref into the select ref-prop

const itemListRenderer: ItemListRenderer<T> = ({items, itemsParentRef, query, renderItem}) => {
	const renderedItems = items.map(renderItem).filter(item => item != null);
        // Accessing the private internal queryList field.
	const queryList = selectRef.current?.queryList;
        // This line is copied from the default implementation, the functions are private/internal as well
	const createItemView = queryList?.isCreateItemRendered() ? queryList.renderCreateItemMenuItem(query.trim()) : null;
	return (
		<Menu ulRef={itemsParentRef}>
			{renderedItems}
			{createItemView}
		</Menu>
	);
};
```

**Comment by gigerdo** [comment]
I think it might be useful to add this feature the same way as renderItem. Since create-item is a core-feature of QueryList, this does not seem like the wrong place to put it.

Here is a pseudo-code changelist on the changes necessary:

itemListRenderer.tsx:
```
export interface IItemListRendererProps<T> {
   //.... existing interface

    renderCreateItem: () => JSX.Element | null;
}
```

queryList.tsx#218
```
// Additional prop, the code is taken from the default renderer
renderCreateItem: () => {
    return this.isCreateItemRendered()
        ? this.renderCreateItemMenuItem(this.state.query.trim())
        : null;
}
```

Example custom renderer / and default renderer:
```
private renderItemList = (listProps: IItemListRendererProps<T>) => {
        const { initialContent, noResults } = this.props;

        // omit noResults if createNewItemFromQuery and createNewItemRenderer are both supplied, and query is not empty
        const maybeNoResults = this.isCreateItemRendered() ? null : noResults;
        const menuContent = renderFilteredItems(listProps, maybeNoResults, initialContent);
        const createItemView = listProps.renderCreateItem();
        if (menuContent == null && createItemView == null) {
            return null;
        }
        return (
            <Menu ulRef={listProps.itemsParentRef}>
                {menuContent}
                {createItemView}
            </Menu>
        );
    };
```

If you like this change I can prepare a PR for you to look at. It should not be much more changes than the few lines above.

**Comment by adidahiya** [comment]
sorry for the delay @gigerdo, but good news - it looks like @rhysbrettbowen has tackled your issue with a very similar solution in #4303

**Comment by gigerdo** [comment]
Great, thanks!
